docs/src/content/docs/manual-setup.mdx

```diff
@@ -125,6 +125,6 @@ In the future, we plan to support this use case better to avoid the need for the
 
 ### Use Starlight with SSR
 
-You can use Starlight alongside custom on-demand rendered pages in your project by following the [“On-demand Rendering Adapters”](https://docs.astro.build/en/guides/server-side-rendering/) guide in Astro’s docs.
+Starlight supports [SSR](https://docs.astro.build/en/guides/server-side-rendering/) out of the box using Astro's server adapters.
 
-Currently, documentation pages generated by Starlight are always prerendered regardless of your project's output mode. We hope to be able to support on-demand rendering for Starlight pages soon.
+Contrary to your own pages, documentation pages generated by Starlight are always prerendered by default regardless of your project's output mode. To opt-out of prerendering your Starlight pages, set the [`prerender` config option](/reference/configuration/#prerender) to `false`.
```

Comment:
```suggestion
Contrary to your own pages, documentation pages generated by Starlight are always pre-rendered by default regardless of your project's output mode. To opt-out of pre-rendering your Starlight pages, set the [`prerender` config option](/reference/configuration/#prerender) to `false`.
```

To match the wording from [Astro Docs](https://docs.astro.build/en/guides/server-side-rendering/#enable-on-demand-server-rendering)